code/modules/integrated_electronics/subtypes/power.dm

```diff
@@ -51,26 +51,26 @@
 			if(A.Adjacent(B))
 				if(AM.loc != assembly)
 					transfer_amount *= 0.8 // Losses due to distance.
-
-				if(cell.charge == cell.maxcharge)
-					return FALSE
-
-				if(transfer_amount && assembly.draw_power(amount_to_move)) // CELLRATE is already handled in draw_power()
-					cell.give(transfer_amount * GLOB.CELLRATE)
 				set_pin_data(IC_OUTPUT, 1, cell.charge)
 				set_pin_data(IC_OUTPUT, 2, cell.maxcharge)
 				set_pin_data(IC_OUTPUT, 3, cell.percent())
 				activate_pin(2)
 				push_data()
-				return TRUE
+				if(cell.charge == cell.maxcharge)
+					return FALSE
+
+				if(transfer_amount && assembly.draw_power(amount_to_move)) // CELLRATE is already handled in draw_power()
+					cell.give(transfer_amount * GLOB.CELLRATE)
+					return TRUE
 		else
 			set_pin_data(IC_OUTPUT, 1, null)
 			set_pin_data(IC_OUTPUT, 2, null)
 			set_pin_data(IC_OUTPUT, 3, null)
 			activate_pin(2)
 			push_data()
 			return FALSE
-	return FALSE
+	else
+		return FALSE
```

Comment:
just do `if(!AM)` at the top so you don't have to indent this much and hurt readability of when this line happens